### DISCUSSION TITLE
Easy way to update an airflow.cfg after un update

### DISCUSSION THREAD

**Discussion by ginolegigot** [apache/airflow#31956] [General]
Hello,
At the moment i have a tedious way to compare the new ansible.cfg template (from a new airflow version) in airflow/config_templates/default_airflow.cfg with the current airflow.cfg file i have in order to stay uptodate in terms of config file writing.
Do you have a simpler process ou any ideas to improve my way of updating an airflow.cfg ?

**Comment by potiuk** [comment]
Yes. 

1. Use environment variables
2. Only keep the values that you changed in your config. Remove all the defaults

The 2. option is recommended. I think the root cause of the problem is that you keep the whole config, where you should only keep whatever you changed from the very beginning.

**Reply by ginolegigot** [reply]
interesting it seems like a whole new paradigm for me. Thanks! I'll think about it!

**Reply by potiuk** [reply]
Well. It's always been like that :) . Nothing new. Maybe not explicilty spelled out anywhere, but this is the way you **should** be doing it.

**Reply by ginolegigot** [reply]
yes i was not just aware of that.
Thanks for your time, have a great day!!
